pkg/resources/external_function_state_upgraders.go

```diff
@@ -0,0 +1,78 @@
+package resources
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+)
+
+type v085ExternalFunctionId struct {
+	DatabaseName             string
+	SchemaName               string
+	ExternalFunctionName     string
+	ExternalFunctionArgTypes string
+}
+
+func parseV085ExternalFunctionId(stringID string) (*v085ExternalFunctionId, error) {
+	reader := csv.NewReader(strings.NewReader(stringID))
```

Comment:
For curiosity, is it a different case than `parseV085FunctionId` to parse it with `csv.NewReader` rather than `strings.Split`?

Reply:
It is copy-pasted from the old implementation. This is the best way, to have the compatible parser. I think we should not refactor these old methods if they were used before as counterpart from the old converters.